### ISSUE TITLE
Next/Previous Buttons submit form

### ISSUE THREAD

**Issue by blanco217** [newbreedofgeek/react-stepzilla#54]
Hello,
 I am creating a form using redux-form and I have a custom survey where each page is displaying as a step. The next/previous buttons that are created do not have the type attribute set so when they are clicked within the form the default behavior for a button is to submit. Adding type="button" fixed my issue.

**Comment by mikkelking** [comment]
I found a similar problem, which I traced to the presence of an enclosing form - if the wizard is inside a form, then the buttons will do a submit. Removing the form fixed the problem (I already had type="button")

**Comment by newbreedofgeek** [comment]
Thanks for the feedback guys. As @mikkelking suggested, is removing the enclosing form an acceptable option? Or @blanco217 did you say you forked the library and added your fix? Did it have other side effects?

**Comment by blanco217** [comment]
@newbreedofgeek For my project I am using redux-form with a multi-step form so in my case it is not an option. The change I made so far has no side effects. I may make a pull request soon when I can test further

**Comment by newbreedofgeek** [comment]
@blanco217 ok, so its as simple as adding type="button" to these 2 buttons right? https://github.com/newbreedofgeek/react-stepzilla/blob/master/src/main.js#L322-L335

**Comment by blanco217** [comment]
@newbreedofgeek yea that is all that is needed

**Comment by nickwesselman** [comment]
Came up with a workaround for this in the meantime:

```
import React from 'react';
import ReactDOM from 'react-dom';
import StepZilla from 'react-stepzilla';

// Workaround for https://github.com/newbreedofgeek/react-stepzilla/issues/54
class StepZillaPatched extends StepZilla {
    componentDidMount() {
        const node = ReactDOM.findDOMNode(this);
        const buttons = node.getElementsByTagName('button');
        for (let i = 0; i < buttons.length; ++i) {
            buttons[i].setAttribute('type', 'button');
        }
    }
}

export default StepZillaPatched;
```

**Comment by atiqueansari1987** [comment]
Anyone working on this? If not can I make changes and submit PR?

**Comment by blanco217** [comment]
@atiqueansari1987 I haven't made the change. If you can that would be great.

**Comment by zeel** [comment]
Added PR.

**Comment by altonotch** [comment]
I encountered this error too. Using version react-stepzilla `4.7.2`, redux-form version `7.3.0`.
I solved the problem by implementing a workaround.

```
// somewhere on component
state = {
    readyForSubmit: false,
}

constructor(props) {
    super(props);
    this.handleStepZillaStepChange = this.handleStepZillaStepChange.bind(this);        
    this.handleFormSubmit = this.handleFormSubmit.bind(this);
}

handleFormSubmit(value)  {
      if (this.state.readyForSubmit) {
          console.log("Submitting the form");
          this.props.onFormSubmit(value);  // this is the the method that actually triggers the call to the api
      }
}

handleStepZillaStepChange(step) {
      if (step === SUBMIT_STEP_NUMBER) { // check if we are on the step that should submit
          console.log('unlocking submit');
          this.setState({readyForSubmit: true});
      }
}

render() {
    const {handleSubmit} = this.props;
    const steps = [...];

    return (
            <div className="new-page">
                <form onSubmit={handleSubmit(this.handleFormSubmit)}>
                    <div className='step-progress'>
                        <StepZilla
                            steps={steps}
                            stepsNavigation={false}
                            nextButtonText="Next"
                            backButtonText="Back"
                            nextTextOnFinalActionStep="Save"
                            preventEnterSubmission={true}
                            onStepChange={this.handleStepZillaStepChange}
                        />
                    </div>
                </form>
            </div>
        );
}
```

**Comment by newbreedofgeek** [comment]
Sorry for the delay everyone.. I have (finally) fixed this via PR #135
